src/twisted/words/protocols/irc.py

```diff
@@ -88,7 +88,7 @@ def parsemsg(s):
     trailing = []
     if not s:
         raise IRCBadMessage("Empty line.")
-    if s[0] == ':':
+    if s[0:1] == ':':
```

Comment:
This is never true though in Python 3 if `s` is actually an instance of `bytes`, as `b':'` is not really equal to `':'`. So either prefix the `':'` with the appropriate "b" or use the rather preferable `startswith` method (with its argument b-prefixed as well, of course) as you won't have to slice the `s` bytes. In fact, all string literals at least in this function wrt. `s` should be prefixed with "b" to actually be Python 2/3 compatible. That is of course, if `s` is certainly an instance of `bytes`, but that I don't really know.

On that note, I'd actually use `b' :' in s` instead of `s.find(b' :') != -1`, but that's just me nitpicking. :)